x-pack/plugins/alerting/index.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Root } from 'joi';
+import { init } from './server';
+import mappings from './mappings.json';
+
+export { AlertingPlugin, AlertsClient, AlertExecuteOptions } from './server';
+
+export function alerting(kibana: any) {
+  return new kibana.Plugin({
+    id: 'alerting',
+    configPrefix: 'xpack.alerting',
+    require: ['kibana', 'elasticsearch', 'actions', 'task_manager'],
+    isEnabled(config: any) {
+      return (
+        config.get('xpack.alerting.enabled') === true &&
+        config.get('xpack.actions.enabled') === true &&
```

Comment:
If for some reason actions isn't enabled, seems like we should still have alerting enabled; presumably it would run through it's intervals, but when actions were to be fired, wouldn't fire them (and provide some indication that actions were disabled or such).  Basically, there's probably >some< value to having alerting "running" even if actions are disabled via the plugin being disabled.

Reply:
Problem is with the old platform, if you depend on something that is not enabled, Kibana will fail to start.  The new platform will support optional dependencies, but for now, this code needs to stay.